### ISSUE TITLE
Add Replay Attack Mitigation Feature

### ISSUE THREAD

**Issue by bijoy26** [ricardoprins/decentralhacks#8]
## Replay Attack
> ### Usually, A **replay attack** is when a signed message is reused to claim authorization for a second action. To avoid replay attacks, > for Ethereum. we use a so-called **nonce**, which is the number of transactions sent by an account. 
> ### The smart contract checks if a nonce is used multiple times.

## Replay attacks are addressed in ethereum by following  - 
![rep](https://user-images.githubusercontent.com/42487891/91310275-77aa8100-e7d3-11ea-8c6a-63539ae569d0.png)

## In Matic, replay attack is likely to occur on Ante Handler - 
> ### [**Heimdall**](https://docs.matic.network/docs/contribute/heimdall/antehandler) is the heart of the Matic system. It manages validators, block producer selection, spans, the state-sync mechanism between Ethereum and Matic and other essentials aspects of the system. 
> ### It is the **PoS validator node** that works in consonance with the Staking contracts on Ethereum to enable the PoS mechanism on Matic.
> ### [**Ante handler**](https://docs.matic.network/docs/contribute/heimdall/antehandler) checks and validates the transaction. After the verification, it checks the balance of the sender for enough fees and deduct fees in case of successful transaction inclusion.
> ### According to [Ante Handler definition](https://docs.matic.network/docs/contribute/heimdall/antehandler), it is prone to replay attack which can be addressed as follows - 
![xxx22](https://user-images.githubusercontent.com/42487891/91311400-dd4b3d00-e7d4-11ea-9454-1d373865a6eb.png)

## Using the above information, you job is to properly pass a parameter `sequenceNumber` (which basically the **nonce**) on the **vote() function** and keep a mapping check to ensure the transaction is executed only once
---
## Edit-1 : 
- ### We are addressing **"Replay attack"** by Implementing 'onlyAuditor modifier' 
- ### We keep the issue of addressing **"Cross-contract Replay attack"** for **future work**. 
  ### It can be performed by using **OpenZeppelin library**'s **['Upgradeable contracts'](https://docs.openzeppelin.com/learn/upgrading-smart-contracts)** feature which allows upgrading existing contracts without changing the deployed address.
  ### If we have a storage and proxy contract, it is possible to perform Update functions to the already deployed contracts through openzeppelin-cli.
